lib/Optimizer/Transforms/LoopUnroll.cpp

```diff
@@ -194,6 +195,48 @@ class LoopUnrollPass : public cudaq::opt::impl::LoopUnrollBase<LoopUnrollPass> {
   }
 };
 
+/// After unrolling the loops, there may be duplicate registerName attributes in
+/// use. This pass will assign them unique names by appending a counter.
+class UpdateRegisterNamesPass
+    : public cudaq::opt::impl::UpdateRegisterNamesBase<
+          UpdateRegisterNamesPass> {
+public:
+  using UpdateRegisterNamesBase::UpdateRegisterNamesBase;
+
+  void runOnOperation() override {
+    auto *op = getOperation();
+
+    // First get a count of the number of times a variable is used.
+    std::unordered_map<std::string, int> varNameCounts;
```

Comment:
We should probably be using DenseMap per LLVM style here.

Reply:
Updated in bf41f69f